### ISSUE TITLE
[regression-44.0.1] Basic HTTP auth (username+password) is broken

### ISSUE THREAD

**Issue by ikiK-CRO** [nativefier/nativefier#1219]
After last fix to 44.0.1 im getting: 
```
Unauthorized
This server could not verify that you are authorized to access the document requested. Either you supplied the wrong credentials (e.g., bad password), or your browser doesn't understand how to supply the credentials required.

Additionally, a 401 Unauthorized error was encountered while trying to use an ErrorDocument to handle the request.
```

**On Mac and windows**

 I am using among others that might be related:  
`--basic-auth-password  --basic-auth-username --insecure --ignore-certificate  `

On 43.1.3 and 44.0.0 this is not a problem.

**Comment by ronjouch** [comment]
@TheCleric regression of your events refactor

**Comment by TheCleric** [comment]
Ugh. I'm so tempted to just revert back to 43.1.3 at this point.

**Comment by ronjouch** [comment]
> Ugh. I'm so tempted to just revert back to 43.1.3 at this point.

@TheCleric I understand the frustration; been through similar regression-after-regression streaks too. Well, your call. The refactors you did make sense, though, and this particular one is easy to iterate on (with a local server, or test websites with basic http auth), so it would also make sense to fix than revert.

**Comment by TheCleric** [comment]
> @TheCleric I understand the frustration; been through similar regression-after-regression streaks too. Well, your call. The refactors you did make sense, though, and this particular regression is easy to iterate on (with a local server, or test websites with basic http auth), so it would also make sense to fix than revert.

I'll try to look into this soon. Interestingly is that it's reported as working in 44.0.0

**Comment by ronjouch** [comment]
> Interestingly is that it's reported as working in 44.0.0

@TheCleric yes, this feels like due to https://github.com/nativefier/nativefier/commit/709557c7ab3c81717286fa4fd21d3204d131f010#diff-17e160d5b101ff3fc2e6244359cd30e4d299ad70b8a7721ed43b6784f2dca95dR271-R286 , which is in 44.0.1 but not 44.0.0

---

By the way, for this kind of high-level / Electron-level / impossible to unit test stuff (and we don't have full end-to-end tests working at graphics server level and doing real clicks in a real app), you can consider enriching our `test:manual` smoke test (that I run through before each release). For example, maybe the current test could be done on a site with basic http auth, or maybe it'd be worth to add a second section to the current script that nativefies a basic-http-auth test site.

**Comment by TheCleric** [comment]
> > Interestingly is that it's reported as working in 44.0.0
> 
> @TheCleric yes, this feels like due to [709557c#diff-17e160d5b101ff3fc2e6244359cd30e4d299ad70b8a7721ed43b6784f2dca95dR271-R286](https://github.com/nativefier/nativefier/commit/709557c7ab3c81717286fa4fd21d3204d131f010#diff-17e160d5b101ff3fc2e6244359cd30e4d299ad70b8a7721ed43b6784f2dca95dR271-R286) , which is in 44.0.1 but not 44.0.0
> 

Yeah I think you're right.


> By the way, for this kind of high-level / Electron-level / impossible to unit test stuff (and we don't have full end-to-end tests working at graphics server level and doing real clicks in a real app), you can consider enriching our `test:manual` smoke test (that I run through before each release). For example, maybe the current test could be done on a site with basic http auth, or maybe it'd be worth to add a second section to the current script that nativefies a basic-http-auth test site.

I've also thought about looking into something like [Spectron](https://www.electronjs.org/spectron) for stuff like this.

**Comment by ronjouch** [comment]
Fix released as 44.0.2, available now on npm. Thanks for the report, and thanks to @TheCleric for the fix!

**Comment by ikiK-CRO** [comment]
All good so far, win and mac. 

Thanks guys for all the work. I appreciate it.
